javaagent-exporters/prometheus/src/main/java/io/opentelemetry/javaagent/exporters/prometheus/PrometheusMetricExporterFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.javaagent.exporters.prometheus;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.exporters.prometheus.PrometheusCollector;
+import io.opentelemetry.javaagent.spi.exporter.ExporterConfig;
+import io.opentelemetry.javaagent.spi.exporter.MetricExporterFactory;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.metrics.MeterSdkProvider;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import io.prometheus.client.exporter.HTTPServer;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@AutoService(MetricExporterFactory.class)
+public class PrometheusMetricExporterFactory implements MetricExporterFactory {
+  private static final Logger log = LoggerFactory.getLogger(PrometheusMetricExporterFactory.class);
+
+  private static final String EXPORTER_NAME = "prometheus";
+  private static final String PORT_CONF_PROP_NAME = EXPORTER_NAME + ".port";
+  private static final String HOST_CONF_PROP_NAME = EXPORTER_NAME + ".host";
+  private static final int DEFAULT_PORT = 9464;
+  // The empty address equals to any address
+  private static final String DEFAULT_HOST = "0.0.0.0";
+
+  private final MeterSdkProvider meterSdkProvider = OpenTelemetrySdk.getMeterProvider();
+
+  @Override
+  public MetricExporter fromConfig(ExporterConfig config) {
+    PrometheusCollector.newBuilder()
+        .setMetricProducer(meterSdkProvider.getMetricProducer())
+        .buildAndRegister();
+    try {
+      int port = config.getInt(PORT_CONF_PROP_NAME, DEFAULT_PORT);
+      String host = config.getString(HOST_CONF_PROP_NAME, DEFAULT_HOST);
+      log.info("Creating Prometheus exporter on host: '{}' and port: '{}'", host, port);
+      HTTPServer server = new HTTPServer(host, port);
+      return new PrometheusExporter(server);
```

Comment:
I have switch to this API, however I still don't get any metrics from my APP. 

An alternative solution might be to implement adapter for `MetricProducer` in `PrometheusMetricExporterFactory` that would cache the metrics returned by the `MetricExporter`.

Reply:
There is smth weird going on, I am not able to get metrics from the app to the agent. 

In the debugger I see the metrics being collected (`OpenTelemetrySdk.getMeterProvider().getMetricProducer().collectAllMetrics()`) but the exporter or collector do not get any metrcics.

Reply:
It seems that the app is using a different instance of `OpenTelemetrySdk.getMeterProvider`  than the agent.

Reply:
My issue is that the `opentelemetry-api-beta` does no kick so the calls to the OTel API in the app are noop. Since I have included only the API.

Reply:
Finally, I have managed to get this working. There are two things that I have learned by hard:
1. the app should use only the OTel API
2. I had to rebase and use OTEL 0.9.0-SNAPSHOT API in the app.

Reply:
sorry about that ☹️

I logged #1279 for the first item, and bumped priority for the second item #1043, though that issue will magically go away once we GA since we will support all GA OTel API versions